autoscale/src/main/java/com/sequenceiq/periscope/service/YarnInvocationService.java

```diff
@@ -0,0 +1,138 @@
+package com.sequenceiq.periscope.service;
+
+import static com.sequenceiq.periscope.model.ScalingAdjustmentType.REGULAR;
+import static com.sequenceiq.periscope.model.ScalingAdjustmentType.STOPSTART;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.response.StackV4Response;
+import com.sequenceiq.periscope.domain.Cluster;
+import com.sequenceiq.periscope.domain.LoadAlert;
+import com.sequenceiq.periscope.domain.LoadAlertConfiguration;
+import com.sequenceiq.periscope.model.ScalingAdjustmentType;
+import com.sequenceiq.periscope.model.yarn.YarnScalingServiceV1Response;
+import com.sequenceiq.periscope.monitor.client.YarnMetricsClient;
+import com.sequenceiq.periscope.monitor.evaluator.load.YarnResponseUtils;
+import com.sequenceiq.periscope.monitor.sender.ScalingEventSender;
+import com.sequenceiq.periscope.utils.ClusterUtils;
+import com.sequenceiq.periscope.utils.StackResponseUtils;
+import com.sequenceiq.periscope.utils.TimeUtil;
+
+@Service
+public class YarnInvocationService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(YarnInvocationService.class);
+
+    @Inject
+    private YarnResponseUtils yarnResponseUtils;
+
+    @Inject
+    private StackResponseUtils stackResponseUtils;
+
+    @Inject
+    private YarnMetricsClient yarnMetricsClient;
+
+    @Inject
+    private ClusterService clusterService;
+
+    @Inject
+    private ScalingEventSender eventSender;
+
+    private Cluster cluster;
+
+    private LoadAlert loadAlert;
+
+    private LoadAlertConfiguration loadAlertConfiguration;
+
+    private String policyHostGroup;
+
+    public void pollYarnMetricsAndScaleCluster(Long clusterId, String pollingUserCrn, boolean stopStartEnabled, StackV4Response stackV4Response)
+            throws Exception {
+
+        cluster = clusterService.findById(clusterId);
```

Comment:
Get these as parameters ?

Reply:
Updated to fetch the entity as a parameter and derived subsequent params from the entity